src/main/scala/com/ibm/couchdb/api/Documents.scala

```diff
@@ -37,17 +37,33 @@ class Documents(client: Client, db: String, typeMapping: TypeMapping) {
     server.mkUuid flatMap (create(obj, _))
   }
 
-  def create[D: W](obj: D, id: String): Task[Res.DocOk] = {
+  def create[D: W](obj: D, attachments: Map[String, Req.Attachment]): Task[Res.DocOk] = {
+    server.mkUuid flatMap (create(obj, _, attachments))
+  }
+
+  def create[D: W](obj: D,
+                   id: String,
+                   attachments: Map[String, Req.Attachment] = Map.empty[String, Req.Attachment]
+                  ): Task[Res.DocOk] = {
     val cl = getClassName(obj)
     if (!types.contains(cl))
```

Comment:
should we use TypeMapping.contains(t: Class) instead (that should exist after merging #27)